nats.go

```diff
@@ -3519,23 +3529,26 @@ func (nc *Conn) Close() {
 
 // IsClosed tests if a Conn has been closed.
 func (nc *Conn) IsClosed() bool {
-	nc.mu.Lock()
-	defer nc.mu.Unlock()
-	return nc.isClosed()
+	nc.mu.RLock()
```

Comment:
Again not performance critical so defer ok and I prefer the defer version to be honest.